diff --git a/current/core/server/data/schema/default-settings.json b/current/core/server/data/schema/default-settings.json
index 7c316ab05..c2f905539 100644
--- a/current/core/server/data/schema/default-settings.json
+++ b/current/core/server/data/schema/default-settings.json
@@ -55,7 +55,7 @@
     },
     "site": {
         "title": {
-            "defaultValue": "Ghost",
+            "defaultValue": "Hugo Freire",
             "validations": {
                 "isLength": {
                     "max": 150
@@ -75,7 +75,7 @@
             "type": "string"
         },
         "logo": {
-            "defaultValue": "",
+            "defaultValue": "/content/images/2019/01/logo.png",
             "type": "string"
         },
         "cover_image": {
@@ -83,7 +83,7 @@
             "type": "string"
         },
         "icon": {
-            "defaultValue": "",
+            "defaultValue": "/content/images/2019/01/icon.png",
             "type": "string"
         },
         "accent_color": {
@@ -102,7 +102,7 @@
             "type": "string"
         },
         "timezone": {
-            "defaultValue": "Etc/UTC",
+            "defaultValue": "Europe/Amsterdam",
             "validations": {
                 "isTimezone": true,
                 "isEmpty": false
@@ -110,7 +110,7 @@
             "type": "string"
         },
         "codeinjection_head": {
-            "defaultValue": "",
+            "defaultValue": "<!-- Google Analytics -->\n<script>\n  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){\n  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),\n  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)\n  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');\n\n  ga('create', 'UA-26338127-1', 'auto');\n  ga('send', 'pageview');\n\n</script>",
             "type": "string"
         },
         "codeinjection_foot": {
@@ -208,7 +208,7 @@
     },
     "theme": {
         "active_theme": {
-            "defaultValue": "casper",
+            "defaultValue": "hugo-exec-sh",
             "flags": "RO",
             "type": "string"
         }
diff --git a/current/core/server/services/webhooks/trigger.js b/current/core/server/services/webhooks/trigger.js
index f10789143..0334cf60c 100644
--- a/current/core/server/services/webhooks/trigger.js
+++ b/current/core/server/services/webhooks/trigger.js
@@ -82,6 +82,16 @@ module.exports = (event, model) => {
                             retry: 5
                         };

+                        const secret = webhook.get('secret');
+                        if (secret) {
+                            const timestamp = Date.now();
+                            opts.headers['X-Ghost-Signature'] = require('crypto')
+                                .createHmac('sha256', secret)
+                                .update(`${timestamp}.${reqPayload}`)
+                                .digest('hex');
+                            opts.headers['X-Ghost-Request-Timestamp'] = timestamp;
+                        }
+
                         logging.info(`Triggering webhook for "${webhook.get('event')}" with url "${url}"`);

                         request(url, opts)
diff --git a/current/core/server/web/parent/app.js b/current/core/server/web/parent/app.js
index 6c5b22627..2604210c2 100644
--- a/current/core/server/web/parent/app.js
+++ b/current/core/server/web/parent/app.js
@@ -10,6 +10,24 @@ module.exports = function setupParentApp(options = {}) {
     debug('ParentApp setup start');
     const parentApp = express('parent');

+    // If we are behind a proxy that uses a non-standard forwarded proto header,
+    // we should replace the X-Forwarded-Proto with its value
+    const forwardedProtoHeader = config.get('forwarded_proto_header');
+    if (forwardedProtoHeader) {
+        parentApp.use(function (req, res, next) {
+            if (req.headers[forwardedProtoHeader]) {
+                req.headers['x-forwarded-proto'] = req.headers[forwardedProtoHeader];
+                req.headers['x-forwarded-port'] = req.headers[forwardedProtoHeader] === 'https' ? 443 : 80;
+            }
+            next();
+        });
+    }
+
+    // Make a ping endpoint available that always returns 200 OK
+    parentApp.get('/ping', function (req, res) {
+        res.send({answer: 'pong'});
+    });
+
     parentApp.use(mw.requestId);
     parentApp.use(mw.logRequest);

diff --git a/current/core/server/web/parent/middleware/log-request.js b/current/core/server/web/parent/middleware/log-request.js
index 121b63d38..6e928aed3 100644
--- a/current/core/server/web/parent/middleware/log-request.js
+++ b/current/core/server/web/parent/middleware/log-request.js
@@ -7,13 +7,15 @@ module.exports = function logRequest(req, res, next) {
     const startTime = Date.now();

     function logResponse() {
-        res.responseTime = (Date.now() - startTime) + 'ms';
+        res.responseTime = (Date.now() - startTime);
         req.userId = req.user ? (req.user.id ? req.user.id : req.user) : null;

         if (req.err && req.err.statusCode !== 404) {
-            logging.error({req: req, res: res, err: req.err});
-        } else {
-            logging.info({req: req, res: res});
+            logging.error(req.err);
+        }
+
+        if (req.headers['user-agent'] !== 'Amazon CloudFront') {
+            logging.debug(`${req.ips || req.ip} - ${req.userId || '-'} [${new Date().toISOString()}] "${req.method} ${req.originalUrl}" HTTP/${req.httpVersionMajor}.${req.httpVersionMinor} ${res.statusCode} ${res.getHeader('content-length') || (res.__headers && res.__headers['Content-Length']) || '-'} "${req.get('referer') || '-'}" "${req.get('user-agent') || '-'}" ${res.responseTime}`);
         }

         res.removeListener('finish', logResponse);
diff --git a/current/loggingrc.js b/current/loggingrc.js
index d3b64d848..2b047d902 100644
--- a/current/loggingrc.js
+++ b/current/loggingrc.js
@@ -1,15 +1,15 @@
-const config = require('./core/shared/config');
+const Logger = require('modern-logger');

-module.exports = {
-    name: config.get('logging:name'),
-    env: config.get('env'),
-    path: config.get('logging:path') || config.getContentPath('logs'),
-    domain: config.get('url'),
-    mode: config.get('logging:mode'),
-    level: config.get('logging:level'),
-    transports: config.get('logging:transports'),
-    gelf: config.get('logging:gelf'),
-    loggly: config.get('logging:loggly'),
-    elasticsearch: config.get('logging:elasticsearch'),
-    rotation: config.get('logging:rotation')
-};
+Logger.configure({
+    transports: {
+        console: [
+            {
+                colorize: false,
+                timestamp: false
+            }
+        ]
+    },
+    enableEmoji: false
+});
+
+module.exports = Logger;
