diff --git a/current/core/server/web/parent-app.js b/current/core/server/web/parent-app.js
index 1673792d1..143a8b62e 100644
--- a/current/core/server/web/parent-app.js
+++ b/current/core/server/web/parent-app.js
@@ -13,6 +13,25 @@ module.exports = function setupParentApp(options = {}) {

     // ## Global settings

+    // If we are behind a proxy that uses a non-standard forwarded proto header,
+    // we should replace the X-Forwarded-Proto with its value
+    var forwardedProtoHeader = config.get('forwarded_proto_header');
+    if (forwardedProtoHeader) {
+        parentApp.use(function (req, res, next) {
+            if (req.headers[forwardedProtoHeader]) {
+                req.headers['x-forwarded-proto'] = req.headers[forwardedProtoHeader];
+                req.headers['x-forwarded-port'] = req.headers[forwardedProtoHeader] === 'https' ? 443 : 80;
+            }
+
+            next();
+        });
+    }
+
+    // Make a ping endpoint available that always returns 200 OK
+    parentApp.get('/ping', function (req, res) {
+        res.send({answer: 'pong'});
+    });
+
     // Make sure 'req.secure' is valid for proxied requests
     // (X-Forwarded-Proto header will be checked, if present)
     parentApp.enable('trust proxy');
diff --git a/current/core/server/web/shared/middlewares/log-request.js b/current/core/server/web/shared/middlewares/log-request.js
index c536cd816..1cc7b5a13 100644
--- a/current/core/server/web/shared/middlewares/log-request.js
+++ b/current/core/server/web/shared/middlewares/log-request.js
@@ -17,7 +17,9 @@ module.exports = function logRequest(req, res, next) {
         if (req.err && req.err.statusCode !== 404) {
             common.logging.error({req: req, res: res, err: req.err});
         } else {
-            common.logging.info({req: req, res: res});
+            if (req.headers['user-agent'] !== 'Amazon CloudFront') {
+                common.logging.info({req: req, res: res});
+            }
         }

         res.removeListener('finish', logResponse);
